functions/transferChatResolve.ts

```diff
@@ -0,0 +1,125 @@
+import '@twilio-labs/serverless-runtime-types';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
+import {
+  responseWithCors,
+  bindResolve,
+  error400,
+  error500,
+  success,
+} from 'tech-matters-serverless-helpers';
+
+const TokenValidator = require('twilio-flex-token-validator').functionValidator;
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  TWILIO_CHAT_TRANSFER_WORKFLOW_SID: string;
+  CHAT_SERVICE_SID: string;
+};
+
+export type Body = {
+  closeSid?: string;
+  keepSid?: string;
+  kickMember?: string;
+  newStatus?: string;
+};
+
+export const handler: ServerlessFunctionSignature = TokenValidator(
+  async (context: Context<EnvVars>, event: Body, callback: ServerlessCallback) => {
+    const client = context.getTwilioClient();
+
+    const response = responseWithCors();
+    const resolve = bindResolve(callback)(response);
+
+    const { closeSid, keepSid, kickMember, newStatus } = event;
+
+    try {
+      if (closeSid === undefined) {
+        resolve(error400('closeSid'));
+        return;
+      }
+      if (keepSid === undefined) {
+        resolve(error400('keepSid'));
+        return;
+      }
+      if (kickMember === undefined) {
+        resolve(error400('kickMember'));
+        return;
+      }
+
+      // retrieve attributes of the closing task
+      const closingTask = await client.taskrouter
+        .workspaces(context.TWILIO_WORKSPACE_SID)
+        .tasks(closeSid)
+        .fetch();
+
+      const closingAttributes = JSON.parse(closingTask.attributes);
+      const keepingChannel = closingAttributes.channelSid;
+
+      // Set the channelSid and ProxySessionSID to a dummy value. This keeps the session alive
+      const updatedClosingAttributes = {
+        ...closingAttributes,
+        channelSid: 'CH00000000000000000000000000000000',
+        proxySessionSID: 'KC00000000000000000000000000000000',
+      };
+
+      await client.taskrouter
+        .workspaces(context.TWILIO_WORKSPACE_SID)
+        .tasks(closeSid)
+        .update({ attributes: JSON.stringify(updatedClosingAttributes) });
+
+      // Close the Task and set the propper status
+      const closedTask = await client.taskrouter
```

Comment:
If the channel is updated in the same call that we are closing the task, the original channel (the one with the contact) is closed, so that's why I've separated them.